frontend/src/app/services/animal.service.ts

```diff
@@ -18,8 +18,12 @@ export class AnimalApiService {
     return this.http.get<Animal[]>(this.baseURL, {responseType: 'json'});
   }
 
-  filterByNameContainsValue(value: string) {
-    return this.http.get<Animal[]>(this.baseURL + value, {responseType: 'json'});
+  filter(param, value) {
```

Comment:
What if you will need filter by several options (for example, by class and name) at the same time?